test/e2e/operator_test.go

```diff
@@ -278,6 +278,7 @@ func TestMutation(t *testing.T) {
 				Image:           "kubernetes/pause",
 				Ports:           []corev1.ContainerPort{{ContainerPort: 80}},
 			}},
+			RestartPolicy: corev1.RestartPolicyNever,
```

Comment:
I should add a test case checking not setting ActiveDeadlineSeconds when RestartPolicy is Always